wreck: convert wrexecd to jansson

grondo opened the pull request:
The main intent of this PR is a quick conversion of `wrexecd` to jansson.

Along the way, the unused per-task `procdesc` information was removed. If we need similar functionality in the future we can implement something more scalably. The `pdesc` JSC tests were removed, but the code from jstatctl.c isn't touched since I know @garlick is working in there.

This was quick work, so could use a good review.

grondo commented:
Hm, getting failures due to exit_status aggregator failures. I'll have to see if this is related to this PR or the aggregator work.

dongahn commented:
Yeah at some point, we will still need something like this to support legacy parallel debuggers etc.

grondo commented:
> Hm, getting failures due to exit_status aggregator failures.

Ugh, hit by `json_pack()` `I` vs `i` again. Fixed now, and rebased against current master.

garlick commented on src/modules/wreck/wrexecd.c:
```diff
@@ -788,13 +789,13 @@ int json_array_to_argv (struct prog_ctx *ctx,
             return (-1);
         }
-        (*argvp) [i] = strdup (json_object_get_string (ox));
+        (*argvp) [i] = strdup (json_string_value (ox));
```
use xstrdup() or check result for null

grondo commented:
Went ahead and added a commit to fix up unchecked calls to `strdup`.

garlick commented:
Thanks!

grondo commented:
Thanks!
